exams/pearson/audit.py

```diff
@@ -0,0 +1,265 @@
+"""Exam auditing"""
+from contextlib import contextmanager
+import logging
+import os
+import re
+
+from boto.s3 import (
+    connection,
+    key,
+)
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+import gnupg
+
+SSE_ENCRYPTION_ALGORITHM = 'AES256'
+REQUIRED_SETTINGS = [
+    'EXAMS_AUDIT_ENCRYPTION_FINGERPRINT',
+    'EXAMS_AUDIT_ENCRYPTION_PUBLIC_KEY',
+]
+
+log = logging.getLogger(__name__)
+
+
+class S3AuditStorage:
```

Comment:
I wanted to have the option of swapping storage mechanisms down the road, a class-based design makes this easier.